### PR TITLE
Migrate some more pallets to construct_runtime

### PR DESCRIPTION
**Pull request by ascjones**
Part of #7949. Migrates:

- babe
- balances
- indices

### CODE REVIEW

File: frame/babe/src/mock.rs [hunk 1]
```diff
@@ -79,7 +85,7 @@ impl frame_system::Config for Test {
 	type AccountId = DummyValidatorId;
 	type Lookup = IdentityLookup<Self::AccountId>;
 	type Header = Header;
-	type Event = ();
+	type Event = Event;
 	type BlockHashCount = BlockHashCount;
 	type PalletInfo = ();
```
Comment (thiolliere) [inline frame/babe/src/mock.rs:90]:
```suggestion
	type PalletInfo = PalletInfo;
```

File: frame/balances/src/tests.rs [hunk 2]
```diff
@@ -48,14 +34,13 @@ macro_rules! decl_tests {
...
 
 		const ID_1: LockIdentifier = *b"1       ";
 		const ID_2: LockIdentifier = *b"2       ";
 
-		pub type System = frame_system::Module<$test>;
-		pub type Balances = Module<$test>;
-
-		pub const CALL: &<$test as frame_system::Config>::Call = &$crate::tests::CallWithDispatchInfo;
+		pub const CALL: &<$test as frame_system::Config>::Call =
+			&Call::Balances(pallet_balances::Call::transfer(0, 0));
```
Comment (ascjones) [inline frame/balances/src/tests.rs:43]: This now needs to be an instance of the actual `Call` from `construct_runtime`, so just constructing a dummy `transfer`. Any better idea?

File: frame/indices/src/mock.rs [hunk 3]
```diff
@@ -106,6 +104,6 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
 	t.into()
 }
 
-pub type System = frame_system::Module<Test>;
-pub type Balances = pallet_balances::Module<Test>;
-pub type Indices = Module<Test>;
+// pub type System = frame_system::Module<Test>;
+// pub type Balances = pallet_balances::Module<Test>;
+// pub type Indices = Module<Test>;
```
Comment (thiolliere) [inline frame/indices/src/mock.rs:109]:
```suggestion
```

**Review by thiolliere — APPROVED** [review]

File: frame/babe/src/mock.rs (hunk 1 above)
Comment (ascjones) [reply frame/babe/src/mock.rs:90]: Damn that was the whole point!

File: frame/babe/src/mock.rs [hunk 4]
```diff
@@ -37,23 +37,29 @@ use sp_consensus_babe::{AuthorityId, AuthorityPair, SlotNumber};
...
-	pub enum Origin for Test where system = frame_system {}
-}
-
-impl_outer_dispatch! {
-	pub enum Call for Test where origin: Origin {
-		babe::Babe,
-		staking::Staking,
-	}
-}
+use pallet_session::{historical as pallet_session_historical};
```
Comment (bkchr) [inline frame/babe/src/mock.rs:40]:
```suggestion
use pallet_session::historical as pallet_session_historical;
```

File: frame/balances/src/tests.rs [hunk 5]
```diff
@@ -48,14 +34,13 @@ macro_rules! decl_tests {
 		};
 		use pallet_transaction_payment::{ChargeTransactionPayment, Multiplier};
 		use frame_system::RawOrigin;
+		use frame_system as system;
```
Comment (bkchr) [inline frame/balances/src/tests.rs:37]: Why do we need this?

**Review by bkchr — APPROVED** [review]

File: frame/balances/src/tests.rs (hunk 5 above)
Comment (ascjones) [reply frame/balances/src/tests.rs]: We don't. Removed.

File: frame/balances/src/tests.rs (hunk 2 above)
Comment (dvdplm) [reply frame/balances/src/tests.rs:42]: I don't have a better idea, but a comment in the code would be welcome. :)

**Pull request merged**